features/keychain/module/crypto/tweak.js

```diff
@@ -0,0 +1,24 @@
+import ecc from '@exodus/bitcoinerlab-secp256k1'
+import { utils } from '@noble/secp256k1'
+
+const tapTweakHash = (publicKey, h) => {
+  const xOnlyPoint = ecc.xOnlyPointFromPoint(publicKey)
+  const hash = utils.taggedHashSync('TapTweak', Buffer.concat(h ? [xOnlyPoint, h] : [xOnlyPoint]))
+  return Buffer.from(hash)
+}
```

Comment:
Can we generalize this and perform `tapTweakHash` by the consumer and just have it pass a buffer to be added to the private key?

* More generalizable API for tweaking keys (Taproot uses TapTweak hash as input but this signer servers more than just bitcoin)
* I'm not super fond of pulling in a complete secp256k1 library just to generate a hash.
*

Reply:
Yes, tapTweakHash can be pulled out. Although we'll still pull in `@noble/secp256k1` because  `@exodus/bitcoinerlab-secp256k1` does.

Reply:
https://github.com/ExodusMovement/exodus-oss/pull/90/commits/867c7d7fde699a04bc8af4a1ee854478e20df6dd